examples/spa/associative_memory.ipynb

```diff
@@ -0,0 +1,462 @@
+     "input": [
+      "input_net = sim.data[input_probe]\n",
+      "output_net = sim.data[output_probe]\n",
+      "print(\"Input dimensions: (%d, %d)\" % input_net.shape)\n",
+      "print(\"Output dimensions: (%d, %d)\" % output_net.shape)"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {},
+     "source": [
+      "We see that for each simulation step (rows of the `input_net` and the `output_net` arrays) there is a 64-dimensional pattern represented in the associative memory. \n",
+      "We use a dot product (implemented in `nengo.spa.similarity`) to compare the similarity of inputs and outputs to all patterns in the vocabulary."
+     ]
+    },
+    {
+     "cell_type": "code",
+     "collapsed": false,
+     "input": [
+      "def plot_similarities(input_net, output_net, vocab1, vocab2=None, top=None):\n",
+      "    if vocab2 is None:\n",
+      "        vocab2 = vocab1\n",
+      "    autoscale=False\n",
+      "    \n",
+      "    if top is None:\n",
+      "        autoscale=True\n",
+      "        \n",
+      "    plt.figure(figsize=(12, 4))\n",
+      "    plt.subplot(1, 2, 1)\n",
+      "    plt.autoscale(autoscale, axis='y')\n",
+      "    plt.plot(t, nengo.spa.similarity(input_net, vocab1), lw=2)\n",
+      "    plt.title(\"Input similarity\")\n",
+      "    plt.xlabel('Time')\n",
+      "    if not autoscale:\n",
+      "        plt.ylim(top=top)\n",
+      "    plt.legend(vocab1.keys, loc='best')\n",
+      "    plt.subplot(1, 2, 2)\n",
+      "    plt.plot(t, nengo.spa.similarity(output_net, vocab2), lw=2)\n",
+      "    plt.title(\"Output similarity\")\n",
+      "    plt.xlabel('Time')\n",
+      "    plt.legend(vocab2.keys, loc='best')\n",
+      "    \n",
+      "plot_similarities(input_net, output_net, vocab, top=1.1)\n"
+     ],
+     "language": "python",
+     "metadata": {},
+     "outputs": []
+    },
+    {
+     "cell_type": "markdown",
+     "metadata": {
+      "collapsed": true
+     },
+     "source": [
+      "We see that the network input has been constant throughout the simulation (`APPLE=1`). Notice that there might be some small similarity between the pattern `APPLE` and some other patterns. This is because the semantic pointers are not orthogonal and the dot product will amount to some value different from zero. This can be changed by increasing the dimensionality of vectors, yielding more dissimilar representations.\n",
+      "At the output, the similarity of the represented semantic pointer with the semantic pointer `APPLE` increases until it reaches the maximal value (`=1`). This means that the associative memory successfully retrieved a pattern similar to the input pattern. The exponential increase is due to the synaptic filtering in `nengo.Probe`. This is used to show how the input to another group of neurons connected to the output of this particular AM module would look like.\n"
```

Comment:
`a pattern similar to the input pattern`
Isn't it almost exactly the input pattern?